src/vos/vos_dtx.c

```diff
@@ -656,9 +672,11 @@ vos_dtx_alloc(struct umem_instance *umm, struct dtx_handle *dth)
 	cont_df = cont->vc_cont_df;
 	dth->dth_gen = cont->vc_dtx_resync_gen;
 
-	D_ALLOC_PTR(dae);
-	if (dae == NULL)
-		return -DER_NOMEM;
+	dae = lrua_allocx(cont->vc_dtx_array, &idx, dth->dth_epoch);
```

Comment:
It is also possible that the array is not full, but we failed to allocate one entry (array_alloc_one). So we may need explicitly return the error code.

Reply:
yeah, possible we want to distinguish between NOMEM and full.  I can rework it